xskillscore/tests/test_probabilistic.py

```diff
@@ -467,29 +467,162 @@ def test_rps_reduce_dim(o, f_prob, category_edges, dim, fair_bool):
+            observations.expand_dims(member_dim),
+            bins=[category_edges],
+            bin_names=["category_edge"],
+            dim=[member_dim],
+        )
+    if "category_edge_bin" in observations_bins.dims:
+        observations_bins = observations_bins.rename(
+            {"category_edge_bin": "category_edge"}
+        )
+
+    forecasts = histogram(
+        forecasts,
+        bins=[category_edges],
+        bin_names=["category_edge"],
+        dim=[member_dim],
+    )
+    if "category_edge_bin" in forecasts.dims:
+        forecasts = forecasts.rename({"category_edge_bin": "category_edge"})
+
+    # normalize f.sum()=1 to make cdf
+    forecasts = forecasts / forecasts.sum(bin_dim)
+
+    Fc = forecasts.cumsum(bin_dim)
+    Oc = observations_bins.cumsum(bin_dim)
+
+    # RPS formulas
+    if fair:
+        Ec = Fc * M
+        res = ((Ec / M - Oc) ** 2 - Ec * (M - Ec) / (M ** 2 * (M - 1))).sum(bin_dim)
+    else:
+        res = ((Fc - Oc) ** 2).sum(bin_dim)
+
+    if weights is not None:
+        res = res.weighted(weights)
+
+    res = res.mean(dim, keep_attrs=keep_attrs)
+    # add bin edges as coords
+    res = res.assign_coords(
+        {"forecasts_category_edge": ", ".join(_get_category_bounds(category_edges))}
+    )
+    res = res.assign_coords(
+        {"observations_category_edge": ", ".join(_get_category_bounds(category_edges))}
+    )
+
+    # keep nans and prevent 0 for all nan grids
+    res = _keep_nans_masked(observations, res, dim, ignore=["category_edge"])
+    return res
+
+
 def test_rps_wilks_example():
     """Test with values from Wilks, D. S. (2006). Statistical methods in the
     atmospheric sciences (2nd ed, Vol. 91). Amsterdam ; Boston: Academic Press. p.301.
     """
-    category_edges = np.array([-0.01, 0.01, 0.24, 10])
+    category_edges = np.array([0.0, 0.01, 0.24, 1.0])
     # first example
-    Obs = xr.DataArray([0.0001])  # no precip
-    F1 = xr.DataArray([0] * 2 + [0.1] * 5 + [0.3] * 3, dims="member")
-    F2 = xr.DataArray([0] * 2 + [0.1] * 3 + [0.3] * 5, dims="member")
-    np.testing.assert_allclose(rps(Obs, F1, category_edges), 0.73)
-    np.testing.assert_allclose(rps(Obs, F2, category_edges), 0.89)
+    # xhistogram way with np.array category_edges
+    Obs = xr.DataArray([0.0001])  # .expand_dims('time')  # no precip
+    F1 = xr.DataArray(
+        [0] * 2 + [0.1] * 5 + [0.3] * 3, dims="member"
+    )  # .expand_dims('time')
+    F2 = xr.DataArray(
+        [0] * 2 + [0.1] * 3 + [0.3] * 5, dims="member"
+    )  # .expand_dims('time')
+    np.testing.assert_allclose(rps_xhist(Obs, F1, category_edges), 0.73)
+    np.testing.assert_allclose(rps_xhist(Obs, F2, category_edges), 0.89)
+    # xr way with xr.DataArray category_edges
+    xr_category_edges = xr.DataArray(
+        category_edges, dims="category_edge", coords={"category_edge": category_edges}
+    )
+    assert_allclose(rps(Obs, F1, category_edges), rps(Obs, F1, xr_category_edges))
```

Comment:
are you missing a test of assert `rps(Obs, F1, category_edges) == 0.73`?. I see it's a test to  ensure the input `rps` with `category_edges` gives the same output of using `xr_category_edges`. I see there is a test above for assert `rps_xhist(Obs, F1, category_edges) ==  0.7`. I know it may be duplicating with the `xhist` way but it can help with testing

Reply:
Ignore. Found that test later.